manifests/plugin/write_http.pp

```diff
@@ -11,6 +17,21 @@
     fail('Only one of nodes or urls is supposed to be defined')
   }
 
+  if $manage_package !~ Undef {
+    $_manage_package = $manage_package
+  } else {
+    if $facts['os']['family'] == 'RedHat' and versioncmp($facts['os']['release']['major'],'8') >= 0 {
```

Comment:
We could get rid of the Optional if this logic is in the params.pp and $manage_packages uses a variable from params.pp?

Reply:
Yes that would work however currently none of the per plugin class inherit from init, params.

We would explode the number of parameters on init.pp if we went down that road.

Would make a lot of sense to switch to module level hiera to achieve this if it were done.

Reply:
Basically there is a matrix of OS vs plugins which you need to keep somewhere.

The generall direction of travel is that RedHat is splitting package while Debian remains as one package so it would at least become a lot simpler in the future so adding 
plugin setting per package now would be a problem for the future.

Reply:
Did you try to access a variable from params.pp? each plugin includes collectd and that inherits from params so this should work?

Reply:
I think it will work, I am just questioning if it is a good idea.

We would need to use `collectd::plugin_http_write_manage_package` and another
one for every plugin, or least where is needs to change based on OS.

And if we go further `collectd::plugin_systemd_package_provide` to cope where
we have rpms on RedHat and pips on debian.

Reply:
One more thing to add.
Such a change would be a major rework of the module, certainly outside the scope of this
bug fix. The `undef/true/false` system is used all over the place currently.


Shall I  create an issue. **Move all OS logic out to module level hiera data** 
With module hiera data we could just define default values for 
`collectd::plugin::http_write` per OS and not pollute init with per metric parameters.

Reply:
And a sub question.

What's the preference for the case where a parameter has per OS defaults in hiera data?
Just define in class as 
```puppet
Boolean $manage_package,
```
with no default

Reply:
my opinion here:
* Data that's the same across all operating systems should not be in module level hiera nor params.pp. It should be directly in the related class
* Everything thats dynamic and can be calcuated within data in modules should go there.

It would be awesome if you could raise an Issue/PR for data in modules. If the data is coming from data in modules the parameter should have no default within the class.

Reply:
Issue created https://github.com/voxpupuli/puppet-collectd/issues/921

Will try some time to look, definitely a worth while task.

@bastelfreak thanks for comments/review.